x-pack/plugins/apm/public/components/shared/Links/DiscoverLinks/__test__/DiscoverLink.test.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import * as savedObject from 'x-pack/plugins/apm/public/services/rest/savedObjects';
+import { mountWithStore } from 'x-pack/plugins/apm/public/utils/testHelpers';
+import { DiscoverLink } from '../DiscoverLink';
+
+describe('DiscoverLink', () => {
+  it('should apply indexPattern to url', async () => {
+    const spy = jest
+      .spyOn(savedObject, 'getAPMIndexPattern')
+      .mockResolvedValue({ id: 'myIndexPatternId' });
+
+    const state = { location: {} };
+    const wrapper = mountWithStore(
+      <DiscoverLink query={{}}>View in Discover</DiscoverLink>,
+      state
+    );
+
+    // TODO: This is nasty - how to wait for render-prop to re-render with async index pattern?
```

Comment:
With enzyme, u should be able to get `wrapper.instance()` and synchronously call the function which is normally asynchronously triggered. It's not the best, since some of the implementation leaks into the test, but it wouldn't be as flaky as setTimeout.

Reply:
Thanks for the suggestion! In this case, `wrapper.instance()` will return `null` since  `DiscoverLink` is a stateless component, right?